### PR TITLE
Support multiple chart lines in `LineChart`

### PR DESCRIPTION
**Pull request by Gowsky**

### CODE REVIEW

**Comment by Gowsky** [comment]
image

File: vico/view/src/main/java/com/patrykandpatryk/vico/view/chart/line/LineChartExtensions.kt [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import com.patrykandpatryk.vico.view.theme.use
+
+/**
+ * Creates a [LineChart.LineSpec] using provided [themeAttrRes] and [styleResId].
+ *
+ * @param context a Context used to retrieve the style information.
+ * @param themeAttrRes a theme attributes resource identifier used to retrieve [LineChart.LineSpec]’s style.
+ * Can be [R.attr.line1spec], [R.attr.line2spec], [R.attr.line3spec], or [0].
+ * @param styleResId used to retrieve the style information if the provided [themeAttrRes] cannot be resolved.
+ * The [styleResId] must define the style of at least one of [R.attr.line1spec], [R.attr.line2spec], [R.attr.line3spec].
```
Comment (patrykmichalik) [inline vico/view/src/main/java/com/patrykandpatryk/vico/view/chart/line/LineChartExtensions.kt:37]:
```suggestion
 * The [styleResId] must define the style of at least one of the following: [R.attr.line1spec], [R.attr.line2spec], and [R.attr.line3spec].
```

File: vico/view/src/main/java/com/patrykandpatryk/vico/view/chart/line/LineChartExtensions.kt [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            typedArray.getNestedTypedArray(
+                context = context,
+                resourceId = 0,
+                styleableResourceId = R.styleable.LineSpec
+            ).getLineSpec(context)
+        }
+}
+
+/**
+ * Creates a [LineChart] using provided [List] of [themeAttrRes] and [styleResId].
```
Comment (patrykmichalik) [inline vico/view/src/main/java/com/patrykandpatryk/vico/view/chart/line/LineChartExtensions.kt:61]:
```suggestion
 * Creates a [LineChart] using the provided [List] of theme attribute resource identifiers and the given [styleResId].
```

File: vico/view/src/main/java/com/patrykandpatryk/vico/view/chart/line/LineChartExtensions.kt [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+ *
+ * @param context a Context used to retrieve the style information.
```
Comment (patrykmichalik) [inline vico/view/src/main/java/com/patrykandpatryk/vico/view/chart/line/LineChartExtensions.kt:63]:
```suggestion
 * @param context the context used to retrieve the style information.
```

File: vico/view/src/main/java/com/patrykandpatryk/vico/view/chart/line/LineChartExtensions.kt [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+ * @param themeAttrRes a [List] of theme attributes resource identifiers used to retrieve [LineChart.LineSpec] styles.
```
Comment (patrykmichalik) [inline vico/view/src/main/java/com/patrykandpatryk/vico/view/chart/line/LineChartExtensions.kt:64]:
```suggestion
 * @param themeAttrRes a [List] of theme attribute resource identifiers used to retrieve [LineChart.LineSpec] styles.
```

File: vico/view/src/main/java/com/patrykandpatryk/vico/view/chart/line/LineChartExtensions.kt [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+ * By default it is a list of [R.attr.line1spec], [R.attr.line2spec], and [R.attr.line3spec].
+ * @param styleResId used to retrieve the style information if the provided [themeAttrRes] cannot be resolved.
+ * The [styleResId] must define the style of at least one of [R.attr.line1spec], [R.attr.line2spec], [R.attr.line3spec].
```
Comment (patrykmichalik) [inline vico/view/src/main/java/com/patrykandpatryk/vico/view/chart/line/LineChartExtensions.kt:67]:
```suggestion
 * The [styleResId] must define the style of at least one of the following: [R.attr.line1spec], [R.attr.line2spec], and [R.attr.line3spec].
```

File: sample/src/main/java/com/patrykandpatryk/vico/sample/compose/component/ComposedChart.kt [hunk 6]
```diff
@@ -50,8 +51,13 @@ internal fun ComposedChart(
     )
 
     val lineChart = lineChart(
-        lineBackgroundShader = lineBackgroundShader,
-        lineColor = MaterialTheme.colorScheme.secondary,
+        lines = listOf(
+            lineSpec(
+                lineBackgroundShader = lineBackgroundShader,
+                lineColor = MaterialTheme.colorScheme.secondary,
+            )
```
Comment (patrykmichalik) [inline sample/src/main/java/com/patrykandpatryk/vico/sample/compose/component/ComposedChart.kt:58]:
```suggestion
            ),
```

File: sample/src/main/java/com/patrykandpatryk/vico/sample/compose/component/ComposedChart.kt [hunk 7]
```diff
@@ -50,8 +51,13 @@ internal fun ComposedChart(
...
+        )
```
Comment (patrykmichalik) [inline sample/src/main/java/com/patrykandpatryk/vico/sample/compose/component/ComposedChart.kt:59]:
```suggestion
        ),
```

File: sample/src/main/java/com/patrykandpatryk/vico/sample/compose/component/ComposedChart.kt [hunk 8]
```diff
@@ -50,8 +51,13 @@ internal fun ComposedChart(
...
+
```
Comment (patrykmichalik) [inline sample/src/main/java/com/patrykandpatryk/vico/sample/compose/component/ComposedChart.kt:60]:
```suggestion
```

File: sample/src/main/java/com/patrykandpatryk/vico/sample/compose/component/LineChartWithLabelsInside.kt [hunk 9]
```diff
@@ -52,7 +53,11 @@ internal fun LineChartWithLabelsInside(
 
     Chart(
         modifier = modifier,
-        chart = lineChart(lineBackgroundShader = null),
+        chart = lineChart(
+            lines = listOf(
+                lineSpec(lineBackgroundShader = null)
```
Comment (patrykmichalik) [inline sample/src/main/java/com/patrykandpatryk/vico/sample/compose/component/LineChartWithLabelsInside.kt:58]:
```suggestion
                lineSpec(lineBackgroundShader = null),
```

File: sample/src/main/java/com/patrykandpatryk/vico/sample/compose/component/LineChartWithLabelsInside.kt [hunk 10]
```diff
@@ -52,7 +53,11 @@ internal fun LineChartWithLabelsInside(
...
+            )
```
Comment (patrykmichalik) [inline sample/src/main/java/com/patrykandpatryk/vico/sample/compose/component/LineChartWithLabelsInside.kt:59]:
```suggestion
            ),
```

File: sample/src/main/java/com/patrykandpatryk/vico/sample/compose/preview/LineChartPreviews.kt [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+
+        Chart(
+            modifier = Modifier.padding(8.dp),
+            chart = lineChart(
+                lines = listOf(
+                    lineSpec(
+                        lineColor = yellow,
+                        lineBackgroundShader = verticalGradient(
+                            arrayOf(yellow.copy(0.5f), yellow.copy(alpha = 0f))
+                        )
```
Comment (patrykmichalik) [inline sample/src/main/java/com/patrykandpatryk/vico/sample/compose/preview/LineChartPreviews.kt:59]:
```suggestion
                        ),
```

File: sample/src/main/java/com/patrykandpatryk/vico/sample/fragment/ViewShowcaseFragment.kt [hunk 12]
```diff
@@ -148,9 +148,13 @@ internal class ViewShowcaseFragment : Fragment() {
...
-                lineColor = secondaryColor
-                lineBackgroundShader = getDottedShader(
-                    dotColor = secondaryColor.copyColor(alpha = Tokens.ComposedChart.SHADER_ALPHA),
+                lines = listOf(
+                    LineChart.LineSpec(
+                        lineColor = secondaryColor,
+                        lineBackgroundShader = getDottedShader(
+                            dotColor = secondaryColor.copyColor(alpha = Tokens.ComposedChart.SHADER_ALPHA),
+                        )
+                    )
```
Comment (patrykmichalik) [inline sample/src/main/java/com/patrykandpatryk/vico/sample/fragment/ViewShowcaseFragment.kt:157]:
```suggestion
                    ),
```

File: vico/view/src/main/java/com/patrykandpatryk/vico/view/chart/line/LineChartExtensions.kt [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+import com.patrykandpatryk.vico.view.R
+import com.patrykandpatryk.vico.view.theme.getLineSpec
+import com.patrykandpatryk.vico.view.theme.getNestedTypedArray
+import com.patrykandpatryk.vico.view.theme.getRawDimension
+import com.patrykandpatryk.vico.view.theme.use
+
+/**
+ * Creates a [LineChart.LineSpec] using provided [themeAttrRes] and [styleResId].
+ *
+ * @param context a Context used to retrieve the style information.
```
Comment (patrykmichalik) [inline vico/view/src/main/java/com/patrykandpatryk/vico/view/chart/line/LineChartExtensions.kt:33]:
```suggestion
 * @param context the context used to retrieve the style information.
```

File: vico/view/src/main/java/com/patrykandpatryk/vico/view/chart/line/LineChartExtensions.kt [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+ * @param themeAttrRes a theme attributes resource identifier used to retrieve [LineChart.LineSpec]’s style.
```
Comment (patrykmichalik) [inline vico/view/src/main/java/com/patrykandpatryk/vico/view/chart/line/LineChartExtensions.kt:34]:
```suggestion
 * @param themeAttrRes a theme attributes resource identifier used to retrieve the [LineChart.LineSpec]’s style.
```

**Review by patrykmichalik — CHANGES_REQUESTED** [review]

**Review by patrykmichalik — APPROVED** [review]
Nice work! 🎉

**Pull request merged**